src/FieldArray.js

```diff
@@ -17,6 +18,10 @@ const createFieldArray = ({ deepEqual, getIn, size }) => {
       this.ConnectedFieldArray = createConnectedFieldArray(context._reduxForm, { deepEqual, getIn, size }, props.name)
     }
 
+    shouldComponentUpdate(nextProps) {
+      return this.props.name !== nextProps.name || shallowCompare(this, nextProps)
```

Comment:
@erikras Yes you're right. `this.props.name !== nextProps.name` would not bring perf wins